avr/cores/megacommand/Midi/MidiSDS.cpp

```diff
@@ -53,6 +53,7 @@ void MidiSDSClass::sendDumpRequest(uint16_t slot) {
 }
 uint8_t MidiSDSClass::waitForMsg(uint16_t timeout) {
 
+  MidiSDSSysexListener.sysex->addSysexListener(&MidiSDSSysexListener);
```

Comment:
I think this is not the right place to add sysex listener.

MidiSDSSysexListenerClass should already add itself as a listener on setup. The above code would result in listener duplication ?

292 void MidiSDSSysexListenerClass::setup(MidiClass *_midi) { sysex = &(_midi->midiSysex); sysex->addSysexListener(this); }

Reply:
let me test a little bit.
I thought the handler wasn't hit earlier.

Reply:
works :)